### PR TITLE
Docs: 2.0 minor changes

### LINKED ISSUE #5261: [DOCS] update phrasing from readme and docs
## Which page or section is this issue related to?

As proposed by @dvsrepo in general comments here: https://github.com/huggingface/hub-docs/pull/1340

### LINKED ISSUE #5303: [DOCS] Update discord channel refernces
## Which page or section is this issue related to?

We renamed the discord channels from `argilla-general` and `argilla-help` to `argilla-distilabel-general` and `argilla-distilabel-help`.

### PR DESCRIPTION
**Pull request by nataliaElv**
# Description

Closes #5303
Closes #5261

**Type of change**

- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by davidberenstein1957** [comment]
Tests are failing due to https://github.com/huggingface/huggingface_hub/releases/tag/v0.21.0

File: argilla/docs/index.md [hunk 1]
```diff
@@ -1,11 +1,10 @@
 ---
-description: Argilla is a **collaboration platform for AI engineers and domain experts** that require **high-quality outputs, full data ownership, and overall efficiency**.
-hide: navigation
+description: Argilla is a collaboration tool for AI engineers and domain experts who need to build high-quality datasets for their projects.hide: navigation
```
Comment (nataliaElv) [inline argilla/docs/index.md]:
```suggestion
- description: Argilla is a collaboration tool for AI engineers and domain experts who need to build high-quality datasets for their projects.
- hide: navigation
```
The home page of the docs seems broken: This description appears above the title. Even with the suggestion here this isn't fixed.

Comment (nataliaElv) [reply argilla/docs/index.md]: @davidberenstein1957

**Comment by nataliaElv** [comment]
@frascuchon @davidberenstein1957 Some tests seem to be failing after the last commit. Is this safe to merge?

**Pull request merged**